### PR TITLE
ci: skip TAP/JUnit and codecov processing for TAV tests

### LINKED ISSUE #1847: ci: avoid spurious TAV Test warnings from not having TAP output
https://github.com/elastic/apm-agent-nodejs/pull/1845 improves the nightly "TAV Test"
run so instead of "all red" for the separate module TAV tests, we get:

- red for those tests that are actually failing, and
- orange warnings for the rest due to the empty `tap2Junit` output

E.g. https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/PR-1845/2/pipeline

Screen Shot 2020-10-30 at 8 32 01 AM

Or the "Test Results" page:
https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-1845/2/testReport/

Screen Shot 2020-10-30 at 8 32 16 AM

I'd like to resolve the "orange warnings" in this ticket. Then subsequent tickets can deal with all of the actual failing tests.

Ideas:

- Change the Jenkins `generateStep` to not bother with the `tap2Junit` for the "TAV Test" run, because it doesn't generate TAP.
  I'm not sure if that is undesirable for the "Test Results" page which, I'm guessing, is relying on JUnit test format output to summarize.
  If that isn't *necessary*, personally I'm happy without an summarize JUnit test count for the full test run.
- Change the "TAV Test" run to see if it *can* emit TAP output that `tap2Junit` would be happy enough with.
  This might be as easy as dropping the `--quiet` option when invoking `tav`.

Another note:
- There is no codecov info generated for the TAV Test steps, so the `codecov` part of the Jenkins step is wasted time.

### PR DESCRIPTION
**Pull request by trentm**
Fixes elastic/apm-agent-nodejs#1847

### CODE REVIEW

**Comment by trentm** [comment]
jenkins run the module tests for ioredis

**Comment by trentm** [comment]
THis nicely shows green for passing TAV tests now:

Screen Shot 2020-10-30 at 1 05 36 PM

**Comment by trentm** [comment]
jenkins run the module tests for redis

**Comment by trentm** [comment]
And actual failing TAV tests still show red:

Screen Shot 2020-10-30 at 2 17 39 PM

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -380,8 +380,12 @@ def generateStep(Map params = [:]){
           } catch(e){
             error(e.toString())
           } finally {
-            tap2Junit(pattern: '*-output.tap', suffix: 'junit.xml', package: 'Node.js', failNever: true)
-            codecov(repo: env.REPO, basedir: '.', secret: "${CODECOV_SECRET}")
+            // TAV tests run 'npm run test:tav' which does *not* produce TAP
+            // output or code coverage .
+            if (! "${tav}") {
```
Comment (v1v) [inline .ci/Jenkinsfile:385]: NIT: in case `tav` was set with only spaces for whatever reason
```suggestion
            if (!tav?.trim()) {
```

**Comment by trentm** [comment]
jenkins run the module tests for ioredis,redis

File: .ci/Jenkinsfile (hunk 1 above)
Comment (trentm) [reply .ci/Jenkinsfile]: Thanks

**Review by v1v — APPROVED** [review]

**Pull request merged**